### PR TITLE
#10861 Added modf to numpy frontend mathematical functions

### PR DESCRIPTION
**Pull request by AryanSharma9917**

### CODE REVIEW

File: ivy/functional/frontends/numpy/mathematical_functions/arithmetic_operations.py [hunk 1]
```diff
@@ -314,3 +314,20 @@ def _fmod(
     if ivy.is_array(where):
         ret = ivy.where(where, ret, ivy.default(out, ivy.zeros_like(ret)), out=out)
     return ret
+
+@handle_numpy_out
+@handle_numpy_dtype
+@to_ivy_arrays_and_back
+@handle_numpy_casting
+@from_zero_dim_arrays_to_scalar
+def _modf(
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/mathematical_functions/arithmetic_operations.py:323]: Need to add all needed function parameters.
Refer to Documentation: https://numpy.org/doc/1.23/reference/generated/numpy.modf.html

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py [hunk 2]
```diff
@@ -640,3 +640,43 @@ def test_numpy_fmod(
...
+
+# modf
+@handle_frontend_test(
+    fn_tree="numpy.modf",
+    dtypes_values_casting=np_frontend_helpers.dtypes_values_casting_dtype(
+        arr_func=[
+            lambda: helpers.dtype_and_values(
+                available_dtypes=helpers.get_dtypes("numeric"),
+                num_arrays=1,
+                max_dims=1,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py:653]: Some of these arguments aren't supported you might have wrong-typed them kindly check the helpers documentation to know accepted arguments https://lets-unify.ai/ivy/testing/hypothesis_helpers.html

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @AryanSharma9917 

Thanks for contributing to ivy😊
Firstly to have your function fully working you will need to add it to the init file as it is a `ufunc` object refer to example here to know how to import it and initialize it:
import: https://github.com/unifyai/ivy/blob/d12bede7d6ecce1728714a49e5d06bf5a92e37e1/ivy/functional/frontends/numpy/__init__.py#L514

instantiate: https://github.com/unifyai/ivy/blob/d12bede7d6ecce1728714a49e5d06bf5a92e37e1/ivy/functional/frontends/numpy/__init__.py#L667

Other changes that are needed are mentioned in comments below.

Thanks again and happy coding🙂

**Comment by AryanSharma9917** [comment]
@hirwa-nshuti I have made all the changes, so I just simply send the pull request or is there anything to be done while sending the pull request to connect those changes to this pull request?

**Comment by hirwa-nshuti** [comment]
> @hirwa-nshuti I have made all the changes, so I just simply send the pull request or is there anything to be done while sending the pull request to connect those changes to this pull request?

You have to commit new changes to this branch you're working on

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py (hunk 2 above)
Comment (hirwa-nshuti) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_arithmetic_operations.py:653]: Still the following data generation is causing failures would you check on them

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey

The unit tests are still failing would you check on them and fix lint errors as well

Thanks

**Comment by AryanSharma9917** [comment]
> Hey
> 
> The unit tests are still failing would you check on them and fix lint errors as well
> 
> Thanks

@hirwa-nshuti could you please help me out, I am facing a problem in solving these two issues.

**Comment by AryanSharma9917** [comment]
@hirwa-nshuti  Do I need to fill in all the errors in lint or the specific files I have worked on during the contribution?

**Pull request closed without merging**